extends CharacterBody2D


const SPEED = 300.0
const JUMP_VELOCITY = -400.0
@export var ACCELERATION = 10
@export var DEACCELERATION = 10
const GRAV : Vector2 = Vector2(0.0, 200.0)


func _physics_process(delta: float) -> void:
	# Add the gravity.
	if not is_on_floor():
		#velocity += get_gravity() * delta
		velocity += GRAV * delta
	#print(get_gravity())
	# Handle jump.
	if Input.is_action_just_pressed("ui_accept") and is_on_floor():
		velocity.y = JUMP_VELOCITY

	# Get the input direction and handle the movement/deceleration.
	# As good practice, you should replace UI actions with custom gameplay actions.
	var direction := Input.get_axis("ui_left", "ui_right")
	if direction:
		#velocity.x = direction * SPEED
		velocity.x = move_toward(0, direction * SPEED, ACCELERATION)
	else:
		velocity.x = move_toward(velocity.x, 0, DEACCELERATION)

	move_and_slide()
